webapp/src/test/java/com/box/l10n/mojito/monitoring/QuartzPendingJobsReportingTaskTest.java

```diff
@@ -0,0 +1,196 @@
+        RequiredSearch search2 = meterRegistry.get("quartz.pending.jobs")
+                                              .tag("jobGroup", MY_GROUP)
+                                              .tag("jobClass", "QuartzPendingJobsReportingTaskTest$Test2Job");
+
+
+        await.untilAsserted(() -> assertThat(search1.gauge().value()).isEqualTo(1));
+        await.untilAsserted(() -> assertThat(search2.gauge().value()).isEqualTo(1));
+    }
+
+    @Test
+    public void testGaugesCountPendingJobsForDynamicGroup() throws Exception {
+
+        JobDetail job;
+        TriggerBuilder<Trigger> builder;
+        int test1Count = random.nextInt(5) + 2;
+        int emptyCount = random.nextInt(5) + 2;
+
+        for (int i = 0; i < test1Count; i++) {
+            job = newJob(Test1Job.class).withIdentity("Test1Job_" + i, DYNAMIC_GROUP_NAME).build();
+            builder = TriggerBuilder.newTrigger()
+                                    .forJob(job)
+                                    .startAt(Date.from(ZonedDateTime.now().plusHours(i+1).toInstant()));
+            scheduler.scheduleJob(job, builder.build());
+        }
+
+        for (int i = 0; i < emptyCount; i++) {
+            job = newJob(EmptyTestJob.class).withIdentity("EmptyJob_" + i, DYNAMIC_GROUP_NAME).build();
+            builder = TriggerBuilder.newTrigger()
+                                    .forJob(job)
+                                    .startAt(Date.from(ZonedDateTime.now().plusHours(i+1).toInstant()));
+            scheduler.scheduleJob(job, builder.build());
+        }
+
+        task.reportPendingJobs();
+
+        RequiredSearch test1Gauges = meterRegistry.get("quartz.pending.jobs")
+                                                  .tag("jobClass", "QuartzPendingJobsReportingTaskTest$Test1Job")
+                                                  .tag("jobGroup", DYNAMIC_GROUP_NAME);
+
+        RequiredSearch test2Gauges = meterRegistry.get("quartz.pending.jobs")
+                                                  .tag("jobClass", "EmptyTestJob")
+                                                  .tag("jobGroup", DYNAMIC_GROUP_NAME);
+
+        await.untilAsserted(() -> assertThat(test1Gauges.gauges()).isNotEmpty());
+        await.untilAsserted(() -> assertThat(test2Gauges.gauges()).isNotEmpty());
+        await.untilAsserted(() -> assertThat(test1Gauges.gauge().value()).isEqualTo(test1Count));
+        await.untilAsserted(() -> assertThat(test2Gauges.gauge().value()).isEqualTo(emptyCount));
+    }
+
+    @Test
+    public void testFetchResults() throws Exception {
+
+        JobDetail job;
+        TriggerBuilder<Trigger> builder;
+        int test1Count = random.nextInt(5) + 2;
+        int emptyCount = random.nextInt(5) + 2;
+
+        for (int i = 0; i < test1Count; i++) {
+            job = newJob(Test1Job.class).withIdentity("Test1Job_" + i, DYNAMIC_GROUP_NAME).build();
```

Comment:
we've been using 
```
    @Rule
    public TestIdWatcher testIdWatcher = new TestIdWatcher();
```

to isolate test data between run and tests. Should probably used here too?

Reply:
Done